extension/src/testStatusBarSingleton.ts

```diff
@@ -0,0 +1,67 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license.
+
+import { window, ProgressLocation, StatusBarAlignment, StatusBarItem } from "vscode";
+
+import * as Commands from "./commands";
+import { TestLevel, TestStatus, TestSuite } from "./protocols";
+
+export class TestStatusBarSingleton {
+    public static getInstance(): TestStatusBarSingleton {
+        return TestStatusBarSingleton.instance;
+    }
+    private static readonly instance: TestStatusBarSingleton = new TestStatusBarSingleton();
+
+    private statusBarItem: StatusBarItem;
+
+    private constructor() {
+        this.statusBarItem = window.createStatusBarItem(StatusBarAlignment.Left, Number.MIN_VALUE);
+    }
+
+    public dispose() {
+        this.statusBarItem.dispose();
+    }
+
+    public init(action: Thenable<void>): Thenable<void> {
+        return window.withProgress({ location: ProgressLocation.Window }, (p) => {
+            p.report({message: 'Loading tests...'});
+            this.statusBarItem.show();
+            return action.then(null,
+            (reason) => {
+                this.statusBarItem.text = 'Failed to load tests';
+            });
+        });
+    }
+
+    public update(tests: TestSuite[], action: Thenable<void>) {
+        return window.withProgress({ location: ProgressLocation.Window }, async (p) => {
+            p.report({message: 'Running/Debugging test...'});
+            this.statusBarItem.text = 'View test output';

```

Comment:
Now the `View test output` is a new button beside the progress info. Can we just make the progress info a button? By doing so, users can just click the progress to see the logs.

Reply:
And suggest `View test logs...`